src/payment/strategies/moneris/moneris-payment-strategy.ts

```diff
@@ -55,6 +57,7 @@ export default class MonerisPaymentStrategy implements PaymentStrategy {
 
         const paymentMethod = getPaymentMethodOrThrow(payment.methodId);
         const testMode = paymentMethod.config.testMode;
+        const shouldSaveInstrument = payment.paymentData && (payment.paymentData as NonceInstrument).shouldSaveInstrument;
```

Comment:
You should first validate that instrument is in fact a nonce. See https://github.com/bigcommerce/checkout-sdk-js/blob/master/src/payment/is-nonce-like.ts

Reply:
I don't think that will work, the `shouldSaveIntrument` is received when the payment is not yet a nonce instrument and since the `isNonceLike` validates the `nonce` field which is missing from this initial payload it will alway return false in the Moneris scenario then making the `shouldSaveIntrument`always undefined.

Reply:
Just to clarify that, I changed to the `HostedInstrument` Type